### ISSUE TITLE
Prism Xamarin page lifecycle not working after selecting a tab [Bug] 

### ISSUE THREAD

**Issue by BaseCrusher** [PrismLibrary/Prism#2117]
### Description
I was trying to lock the user on a specific tab on the screen. 
What I have done was to use the "await _navigationService.SelectTabAsync("Tab3"); " method, on the onDisapearing event.  So that if the user tries switch the tab, he is directed to this tab again.

But after SelectTabAsync is called, the OnDisappearing isn't called anymore until you reload the page. So the user can still move to another page.

### Steps to Reproduce

1. Create a tabbedpage with two pages
2. To the OnDisappearingevent add the "await _navigationService.SelectTabAsync("Tab3"); " 
3. Start and try to navigate from the tab with the SelectTabAsyncmethod to another one two times in a row.

### Expected Behavior
OnDisappearing to be called when the user navigate to another page after the app called SelectTabAsync.

### Actual Behavior
After SelectTabAsync has switched the page , when the user is switching the page again manually, OnDisappearing is not called.

### Basic Information

- Version with issue: 7.2.0.1422
- Last known good version: --
- Xamarin.Forms version: 4.6.0.726
- IDE: Visual Studio Community 2019 v. 16.5.4

### Reproduction Link
https://github.com/BaseCrusher/PrismBugReport

**Comment by brianlagunas** [comment]
Using your repo to debug your issue, you have introduced a recursive code block. In the OnDisappearing in ViewAViewModel you are selecting the ViewA tab again. This will be recursive and never end because XF will never finish setting the CurrentPage to ViewB, yet will continuously try which will invoke the ViewAViewModel OnDisapperaing each time.

You can test this yourself by eliminating Prism and trying to hard code it in the code behind:
```
        protected override void OnDisappearing()
        {
            base.OnDisappearing();

            TabbedPage tabbedPage = this.Parent.Parent as TabbedPage;
            var child = tabbedPage.Children[0];
            tabbedPage.CurrentPage = child;
        }
```

Still fails.

To put it simply, don't do this.

Also, to answer your overall goal of trying to prevent a tab switch... You don't.  I would argue that if you want to prevent a user from going to another page, then a TabControl is probably not what you want to be using.
